src/ctl/alloc.c

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2019-2024, Intel Corporation */
+
+#include <errno.h>
+
+#include "alloc.h"
+#include <string.h>
+
+// #include "fault_injection.h"
+
+Malloc_func fn_malloc = malloc;
+Realloc_func fn_realloc = realloc;
+
+#if FAULT_INJECTION
+#include "log_internal.h"
+
+static __thread int malloc_num;
+static __thread int fail_malloc_num;
+static __thread const char *fail_malloc_from;
+
+void *_flt_Malloc(size_t size, const char *func) {
+    if (fail_malloc_from && strcmp(func, fail_malloc_from) == 0) {
+        if (++malloc_num == fail_malloc_num) {
+            errno = ENOMEM;
+            return NULL;
+        }
+    }
+    return fn_malloc(size);
+}
+
+static __thread int realloc_num;
```

Comment:
do we need this fault injection? this was part of testing infra.

Reply:
Removed, thanks